### PR TITLE
Add type hints

### PR DESCRIPTION
**Pull request by superstructor**
Adding more type safety esp for consuming projects of the APIs. Using e.g. `Union[Account, int, str]` instead of `Account | int | str` etc so that it should remain compatible with older (3.8?) Python releases.

### CODE REVIEW

**Review by billbsing — CHANGES_REQUESTED** [review]
looks, good but the Lint checks are failing

File: convex_api/utils.py [hunk 1]
```diff
@@ -117,8 +117,10 @@ def add_0x_prefix(text: Union[str, None]) -> Union[str, None]:
     :returns: The text with a `0x` appended to the front
 
     """
-    if text:
```
Comment (billbsing) [inline convex_api/utils.py:141]: `if text:` is testing for text == None and len(text) == 0
`if text is None` is only testing for == None not len(text) == 0

File: convex_api/api.py
Comment (billbsing) [inline convex_api/api.py:1]: You can get rid of the 'language' option, since Convex  in the past we had convex-script but now there is only one language

**Review by billbsing — APPROVED** [review]
looks good

File: convex_api/api.py
Comment (superstructor) [reply convex_api/api.py:1]: Thanks. Removed.

**Pull request merged**